app/controllers/concerns/phone_confirmation_flow.rb

```diff
@@ -84,4 +98,16 @@ def clear_session_data
     user_session.delete(unconfirmed_phone_session_key)
     user_session.delete(confirmation_code_session_key)
   end
+
+  def sms_enabled?
+    current_otp_delivery_method == 'sms'
+  end
+
+  def current_otp_delivery_method
+    [params[:delivery_method]].find { |type| /sms|voice|totp/ =~ type }
```

Comment:
I love a good regex. Hard to tell here though if the comparison is brittle or not. Could this be rewritten w/o the regex? It's hard for me to tell what this method actually does.